support-frontend/assets/pages/[countryGroupId]/checkout.tsx

```diff
@@ -496,6 +496,14 @@ function CheckoutComponent({
 	 */
 	let productFields: RegularPaymentRequest['product'];
 	switch (productKey) {
+		case 'GuardianLight':
+			productFields = {
+				productType: 'GuardianLight',
```

Comment:
I know this is just following the established pattern but it does seem to be ready for some genericness here maybe? The product type is just equal to the product key we switched on

Reply:
Oops I missed this! That's a good spot, yeah maybe there's a refactoring here which could reduce duplication. I'll take a look.

Reply:
> The product type is just equal to the product key we switched on

Looking more closely this is the case for _some_ product keys, but not all. So we could do something like:

```js
productType: productKey,
```

for the cases where it's true, but I wonder if that's got the potential for confusion as we're mixing concepts. Product keys aren't the same as product types, for example the keys `NationalDelivery`, `SubscriptionCard`, `HomeDelivery` all have a product type of `Paper` (so I guess it's a slightly higher level concept), but making this change could lead future readers to think that they _are_ the same thing. What do you think?

Reply:
ah that makes sense. Keep it explicit 👍